lib/queuePopulator/LogReader.js

```diff
@@ -522,6 +484,94 @@ class LogReader {
         return undefined;
     }
 
+    _processLogEntry(batchState, record, entry, cb) {
+        function _executeFilter(ext, entry, cb) {
+            if (typeof ext.filterAsync === 'function') {
+                ext.filterAsync(entry, cb);
+            } else {
+                ext.filter(entry);
+                process.nextTick(cb);
```

Comment:
this should be just `callbackify(ext.filter)`

Reply:
also, wondering if this should not be done just once, e.g. on startup process the list of extenions to build the list of callbacks : then just call them for processing

Reply:
IMO i think that adds unnecessary complexity to the code especially since we are not waiting on any return value